### ISSUE TITLE
Allow a different base image to be used for building

### ISSUE THREAD

**Issue by crondinini** [railwayapp/nixpacks#401]
### Feature request

Allow a different base image to be used during the `build` command.

As far as I understand by looking at the code and reading docs, we cannot currently set a different base image to `build`. 

### Motivation

Be able to host the base image in another container registry.

**Comment by coffee-cup** [comment]
This is an interesting request. Out of curiosity a few questions
- Would the base image be a custom one that you created?
- What differences to the current base image would there be?

Currently the base image has everything required to install Nix packages and run the code for the various providers, which is why we don't allow specifying another base image (since we can can't guarantee that everything will be setup correctly).

**Comment by crondinini** [comment]
@coffee-cup Yes I understand your concern. See below the answers to your questions.

> Would the base image be a custom one that you created? 

It would very similar, if not exactly the same as the one currently used. It's less about the differences of image and more about some limitations around Docker Hub.  Having a custom base image would allows `nixpacks` to run in shared CI envs like CodeBuild without the need to authenticate to Docker Hub by hosting the current base image to ECR. 

> What differences to the current base image would there be?

The current base image has everything so the only difference would be allowing advanced users to specify base images gives them the ability to optimize their environments. Specifying the base image allows advanced use cases and open up for the user to add more software that they might need.

**Comment by coffee-cup** [comment]
That makes sense. A PR for adding the ability to change the base image (with something like a `--base-image` flag) would be very welcome 😄

**Comment by JakeCooper** [comment]
^^Just to add to that. @coffee-cup is currently working on building out a few things more related to "overarching architecture" of nixpacks (pipelines, multiple providers, etc) so there's not really cycles to do this (or even a timeline)

However, we would absolutely take a PR that allows you to define a base image, using the --base-image flag) if you or any of the FlightControl team has cycles!

**Comment by mabadir** [comment]
Thanks @coffee-cup, that will be really helpful.

**Comment by mabadir** [comment]
@coffee-cup this PR is related, your message was the key to figure out the issue :)
https://github.com/railwayapp/nixpacks/pull/567
